Internal/Deploy.ps1

```diff
@@ -15,323 +16,197 @@ $errorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-S
 try {
 
-    if ($github) {
-        if (!$githubOwner -or !$token) { throw "When running deploy in a workflow, you need to set githubOwner and token" }
+    # Authenticate to GIT and GH
+    invoke-git config --global user.email "$githubOwner@users.noreply.github.com"
+    invoke-git config --global user.name "$githubOwner"
+    invoke-git config --global hub.protocol https
+    invoke-git config --global core.autocrlf false
+    $ENV:GITHUB_TOKEN = ''
 
-        invoke-git config --global user.email "$githubOwner@users.noreply.github.com"
-        invoke-git config --global user.name "$githubOwner"
-        invoke-git config --global hub.protocol https
-        invoke-git config --global core.autocrlf false
-        $ENV:GITHUB_TOKEN = ''
-    }
     Write-Host "Authenticating with GitHub using token"
     $token | invoke-gh auth login --with-token
-    if ($github) {
-        $ENV:GITHUB_TOKEN = $token
-    }
+    $ENV:GITHUB_TOKEN = $token
 
+    # All references inside microsoft/AL-Go and forks of it are to microsoft/AL-Go-Actions@main, microsoft/AL-Go-PTE@main and microsoft/AL-Go-AppSource@main
+    # When deploying to new repos, the originalOwnerAndRepo should be changed to the new owner and repo
     $originalOwnerAndRepo = @{
-        "actionsRepo" = "microsoft/AL-Go-Actions"
+        "actionsRepo"            = "microsoft/AL-Go-Actions"
         "perTenantExtensionRepo" = "microsoft/AL-Go-PTE"
-        "appSourceAppRepo" = "microsoft/AL-Go-AppSource"
+        "appSourceAppRepo"       = "microsoft/AL-Go-AppSource"
     }
     $originalBranch = "main"
 
-    Set-Location $PSScriptRoot
-    $baseRepoPath = invoke-git -returnValue rev-parse --show-toplevel
+    $baseRepoPath = $ENV:GITHUB_WORKSPACE
     Write-Host "Base repo path: $baseRepoPath"
+    Set-Location $baseRepoPath
+
+    # Whoami
     $user = invoke-gh api user -silent -returnValue | ConvertFrom-Json
     Write-Host "GitHub user: $($user.login)"
 
-    if ($configName -eq "") { $configName = $user.login }
-    if ([System.IO.Path]::GetExtension($configName) -eq "") { $configName += ".json" }
-    $config = Get-Content $configName -Encoding UTF8 | ConvertFrom-Json
-
-    Write-Host "Using config file: $configName"
+    # Dump configuration
+    Write-Host "Configuration:"
     $config | ConvertTo-Json | Out-Host
 
-    Set-Location $baseRepoPath
-
-    if ($algoBranch) {
-        invoke-git checkout $algoBranch
-    }
-    else {
-        $algoBranch = invoke-git -returnValue branch --show-current
-        Write-Host "Source branch: $algoBranch"
-    }
-    if ($collect) {
-        $status = invoke-git -returnValue status --porcelain=v1 | Where-Object { ($_) -and ($_.SubString(3) -notlike "Internal/*") }
-        if ($status) {
-            throw "Destination repo is not clean, cannot collect changes into dirty repo"
-        }
-    }
+    # Get Source Branch
+    $algoBranch = invoke-git -returnValue branch --show-current
+    Write-Host "Source branch: $algoBranch"

```

Comment:
The source branch should almost always be `main` (and maybe `PPPreview` in some cases), right? Should there be a defensive check to avoid deploying from another branch accidentally?

Reply:
I can still use deploy to deploy AL-Go@mybranch to 3 repos and @mybranch for testing.
That should still be supported.

Reply:
We could however add a check that when using copyToMain - algoBranch needs to be main.
Have done that.